yesod-core/Yesod/Routes/TH/RenderRoute.hs

```diff
@@ -8,26 +8,30 @@ module Yesod.Routes.TH.RenderRoute
     ) where
 
 import Yesod.Routes.TH.Types
+#if MIN_VERSION_template_haskell(2,11,0)
+import Language.Haskell.TH (conT)
+#endif
 import Language.Haskell.TH.Syntax
 import Data.Maybe (maybeToList)
 import Control.Monad (replicateM)
 import Data.Text (pack)
 import Web.PathPieces (PathPiece (..), PathMultiPiece (..))
 import Yesod.Routes.Class
 #if __GLASGOW_HASKELL__ < 710
+import Control.Applicative ((<$>))
 import Data.Monoid (mconcat)
 #endif
 
 -- | Generate the constructors of a route data type.
-mkRouteCons :: [ResourceTree Type] -> ([Con], [Dec])
-mkRouteCons =
-    mconcat . map mkRouteCon
+mkRouteCons :: [ResourceTree Type] -> Q ([Con], [Dec])
+mkRouteCons rttypes =
+    mconcat <$> mapM mkRouteCon rttypes
```

Comment:
Yes, `mkRouteCon` used to be pure (returning `Dec`), but now it needs to run in the `Q` monad (now returning `Q Dec`) so that `mapM conT` can be used in the `MIN_VERSION_template_haskell(2,11,0)` case.